containerruntimediscovery/server/server.go

```diff
@@ -107,13 +107,14 @@ func (crds *ContainerRuntimeDiscoveryServer) ExportImage(c echo.Context) error {
 	ctx := c.Request().Context()
 	id := c.Param("id")
 
-	reader, err := crds.discoverer.ExportImage(ctx, id)
+	reader, cleanup, err := crds.discoverer.ExportImage(ctx, id)
 	if err != nil {
 		if errors.Is(err, types.ErrNotFound) {
 			return echo.NewHTTPError(http.StatusNotFound, fmt.Sprintf("Image not found with ID %v", id))
 		}
 		return fmt.Errorf("failed to export image %s: %w", id, err)
 	}
+	defer cleanup()
 	defer reader.Close()
```

Comment:
this might cause some weird internal side effects due to call order, we should probably make it explicit:
```suggestion
	defer func() {
		_ = reader.Close() // close first as cleanup might depend on reader
		_ = cleanup()
	}
```

Reply:
I copied this part from the code from the ExportContainer function: https://github.com/openclarity/openclarity/blob/6e73397737cbb67423d1492d5c7866a766b092d7/containerruntimediscovery/server/server.go#L170

There is nothing wrong with the call order since deferred functions are executed in last-in-first-out order.

Reply:
The execution is okay, but it's clearer (for the developer) what the actual cleanup logic is when explicit